homeassistant/components/tomorrowio/config_flow.py

```diff
@@ -0,0 +1,214 @@
+            vol.Required(
+                CONF_LONGITUDE,
+                "location",
+                default=input_dict.get(CONF_LONGITUDE, hass.config.longitude),
+            ): cv.longitude,
+        },
+    )
+
+
+def _get_unique_id(hass: HomeAssistant, input_dict: dict[str, Any]):
+    """Return unique ID from config data."""
+    return (
+        f"{input_dict[CONF_API_KEY]}"
+        f"_{input_dict.get(CONF_LATITUDE, hass.config.latitude)}"
+        f"_{input_dict.get(CONF_LONGITUDE, hass.config.longitude)}"
+    )
+
+
+class TomorrowioOptionsConfigFlow(config_entries.OptionsFlow):
+    """Handle Tomorrow.io options."""
+
+    def __init__(self, config_entry: config_entries.ConfigEntry) -> None:
+        """Initialize Tomorrow.io options flow."""
+        self._config_entry = config_entry
+
+    async def async_step_init(self, user_input: dict[str, Any] = None) -> FlowResult:
+        """Manage the Tomorrow.io options."""
+        if user_input is not None:
+            return self.async_create_entry(title="", data=user_input)
+
+        options_schema = {
+            vol.Required(
+                CONF_TIMESTEP,
+                default=self._config_entry.options[CONF_TIMESTEP],
+            ): vol.In([1, 5, 15, 30]),
+        }
+
+        return self.async_show_form(
+            step_id="init", data_schema=vol.Schema(options_schema)
+        )
+
+
+class TomorrowioConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Tomorrow.io Weather API."""
+
+    VERSION = 1
+
+    def __init__(self) -> None:
+        """Initialize config flow."""
+        self._showed_import_message = 0
+        self._import_config: dict[str, Any] | None = None
+
+    @staticmethod
+    @callback
+    def async_get_options_flow(
+        config_entry: config_entries.ConfigEntry,
+    ) -> TomorrowioOptionsConfigFlow:
+        """Get the options flow for this handler."""
+        return TomorrowioOptionsConfigFlow(config_entry)
+
+    async def async_step_user(self, user_input: dict[str, Any] = None) -> FlowResult:
+        """Handle the initial step."""
+        errors = {}
+        if user_input is not None:
+            # Grab the API key and add it to the rest of the config before continuing
+            if self._import_config:
+                self._import_config[CONF_API_KEY] = user_input[CONF_API_KEY]
+                user_input = self._import_config.copy()
+            await self.async_set_unique_id(
+                unique_id=_get_unique_id(self.hass, user_input)
+            )
+            self._abort_if_unique_id_configured()
+
+            latitude = user_input.get(CONF_LATITUDE, self.hass.config.latitude)
+            longitude = user_input.get(CONF_LONGITUDE, self.hass.config.longitude)
+            if CONF_NAME not in user_input:
+                user_input[CONF_NAME] = DEFAULT_NAME
+            if zone_state := async_active_zone(self.hass, latitude, longitude):
+                user_input[
```

Comment:
Is this missing a test?

Reply:
yes, I am not sure why this didn't get caught by CI before since the PR that added this code was much smaller. I need to change this behavior as well, I think we should only append the zone name if we are using the default name for `climacell` or `tomorrowio`

Reply:
will fix this later today